lib/common/rand.c

```diff
@@ -0,0 +1,55 @@
+#include <openssl/rand.h>
+#include <inttypes.h>
+#include "h2o/rand.h"
+#include "h2o/string_.h"
+#include "picotls/openssl.h"
+
+static void format_uuid_rfc4122(char *dst, const uint8_t *uuid)
+{
+    // String Representation:
+    // > UUID  = time-low "-" time-mid "-"
+    // >         time-high-and-version "-"
+    // >         clock-seq-and-reserved
+    // >         clock-seq-low "-" node
+    // See also "4.1.2. Layout and Byte Order" for the layout
+    size_t pos = 0;
+#define UUID_ENC_PART(b, p, u, start, last) do { \
+        h2o_hex_encode(&b[p], &u[start], last - start + 1); \
+        p += (last - start + 1) * 2; \
+    } while (0)
+
+    UUID_ENC_PART(dst, pos, uuid, 0, 3); /* time_low */
+    dst[pos++] = '-';
+    UUID_ENC_PART(dst, pos, uuid, 4, 5); /* time_mid */
+    dst[pos++] = '-';
+    UUID_ENC_PART(dst, pos, uuid, 6, 7); /* time_hi_and_version */
+    dst[pos++] = '-';
+    UUID_ENC_PART(dst, pos, uuid, 8, 8); /* clock_seq_hi_and_reserved */
+    UUID_ENC_PART(dst, pos, uuid, 9, 9); /* clock_seq_low */
+    dst[pos++] = '-';
+    UUID_ENC_PART(dst, pos, uuid, 10, 15); /* node */
+#undef UUID_ENC_PART
+
+    /* '\0' is set by h2o_hex_encode() */
+}
```

Comment:
If you want to have unit test for this function, you can create t/00unit/lib/common/rand.c, much like other files under t/00unit/lib/common.